File: single-language-analysis/dataset-files/java/TestSmell004.java
// Original URL: https://github.com/Glacier173/wgit-project/blob/8820c845d359331793302eed996b34d9c2b017f4/GitTest.java#L43-L60

public class TestSmell004 {
    @Test
    void testWriteToIndex() throws Exception {
        Git g = new Git ();
        g.initialize();
        g.addBlob("a.txt");
        g.writeToIndex();
        BufferedReader reader = new BufferedReader(new FileReader("index"));
            String line;
            HashMap<String, String> hm = g.hm;
            while ((line = reader.readLine()) != null) {
                String[] parts = line.split(" : ");
                assertEquals(2, parts.length);
                String fileName = parts[0];
                String hash = parts[1];
                assertTrue(hm.containsKey(fileName));
                assertEquals(hm.get(fileName), hash);
            }
    }
}

File: single-language-analysis/dataset-files/python/test_smell_004.py
import pytest

class TestGitTest:
    def test_smell_004(self):
        g = Git()
        g.initialize()
        g.add_blob("a.txt")
        g.write_to_index()
        reader = open("index", "r")
        line = reader.readline()
        hm = g.hm
        while line:
            parts = line.split(" : ")
            assert len(parts) == 2
            file_name = parts[0]
            hash = parts[1]
            assert file_name in hm
            assert hm[file_name] == hash
            line = reader.readline()
        reader.close()